From 3ec25b457ceffb9edf551f096a11d7671831f4fa Mon Sep 17 00:00:00 2001
From: Mingcong Bai <baimingcong@uniontech.com>
Date: Fri, 28 Jun 2024 14:41:59 +0800
Subject: [PATCH 088/360] arch: arm64: config: enable all submitted Phytium
 features

Enable all features introduced by Phytium's patches in our defconfig.

Signed-off-by: Mingcong Bai <baimingcong@uniontech.com>
---
 .../configs/deepin_arm64_desktop_defconfig    | 32 +++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/arch/arm64/configs/deepin_arm64_desktop_defconfig b/arch/arm64/configs/deepin_arm64_desktop_defconfig
index 00d0dc28e54a..bd779448f4b2 100644
--- a/arch/arm64/configs/deepin_arm64_desktop_defconfig
+++ b/arch/arm64/configs/deepin_arm64_desktop_defconfig
@@ -761,6 +761,7 @@ CONFIG_PCI_HOST_GENERIC=y
 CONFIG_PCIE_MEDIATEK=m
 CONFIG_PCIE_MEDIATEK_GEN3=m
 CONFIG_PCIE_MT7621=m
+CONFIG_PCIE_PHYTIUM_EP=m
 CONFIG_PCIE_MICROCHIP_HOST=m
 CONFIG_PCI_TEGRA=y
 CONFIG_PCIE_RCAR_HOST=y
@@ -900,6 +901,9 @@ CONFIG_MTD_ONENAND_VERIFY_WRITE=y
 CONFIG_MTD_ONENAND_GENERIC=m
 CONFIG_MTD_ONENAND_OTP=y
 CONFIG_MTD_ONENAND_2X_PROGRAM=y
+CONFIG_MTD_RAW_NAND=m
+CONFIG_MTD_NAND_PHYTIUM_PCI=m
+CONFIG_MTD_NAND_PHYTIUM_PLAT=m
 # CONFIG_MTD_NAND_ECC_MXIC is not set
 CONFIG_MTD_LPDDR=m
 CONFIG_MTD_SPI_NOR=m
@@ -947,6 +951,7 @@ CONFIG_AD525X_DPOT_I2C=m
 CONFIG_AD525X_DPOT_SPI=m
 CONFIG_DUMMY_IRQ=m
 CONFIG_PHANTOM=m
+CONFIG_PHYTIUM_LPC_SNOOP=m
 CONFIG_ICS932S401=m
 CONFIG_ATMEL_SSC=m
 CONFIG_ENCLOSURE_SERVICES=m
@@ -1447,6 +1452,7 @@ CONFIG_SMSC911X=m
 CONFIG_SMSC9420=m
 CONFIG_STMMAC_ETH=y
 CONFIG_DWMAC_DWC_QOS_ETH=m
+CONFIG_DWMAC_PHYTIUM=m
 CONFIG_STMMAC_PCI=y
 CONFIG_HAPPYMEAL=m
 CONFIG_SUNGEM=m
@@ -1514,6 +1520,9 @@ CONFIG_CAN_CC770_PLATFORM=m
 CONFIG_CAN_IFI_CANFD=m
 CONFIG_CAN_M_CAN=m
 CONFIG_CAN_PEAK_PCIEFD=m
+CONFIG_CAN_PHYTIUM=m
+CONFIG_CAN_PHYTIUM_PLATFORM=m
+CONFIG_CAN_PHYTIUM_PCI=m
 CONFIG_CAN_SJA1000=m
 CONFIG_CAN_EMS_PCI=m
 CONFIG_CAN_KVASER_PCI=m
@@ -1796,6 +1805,7 @@ CONFIG_KEYBOARD_XTKBD=m
 CONFIG_KEYBOARD_CAP11XX=m
 CONFIG_KEYBOARD_BCM=m
 CONFIG_KEYBOARD_MTK_PMIC=m
+CONFIG_KEYBOARD_PHYTIUM=m
 CONFIG_MOUSE_PS2_SENTELIC=y
 CONFIG_MOUSE_PS2_TOUCHKIT=y
 CONFIG_MOUSE_SERIAL=m
@@ -2012,6 +2022,7 @@ CONFIG_SERIAL_AMBA_PL010=y
 CONFIG_SERIAL_AMBA_PL010_CONSOLE=y
 CONFIG_SERIAL_AMBA_PL011=y
 CONFIG_SERIAL_AMBA_PL011_CONSOLE=y
+CONFIG_SERIAL_PHYTIUM_PCI=m
 CONFIG_SERIAL_ATMEL=y
 CONFIG_SERIAL_ATMEL_CONSOLE=y
 CONFIG_SERIAL_MESON=m
@@ -2076,6 +2087,8 @@ CONFIG_RPMSG_TTY=m
 CONFIG_SERIAL_DEV_BUS=m
 CONFIG_VIRTIO_CONSOLE=y
 CONFIG_ASPEED_KCS_IPMI_BMC=m
+CONFIG_PHYTIUM_KCS_IPMI_BMC=m
+CONFIG_PHYTIUM_BT_IPMI_BMC=m
 CONFIG_NPCM7XX_KCS_IPMI_BMC=m
 CONFIG_IPMI_KCS_BMC_CDEV_IPMI=m
 CONFIG_IPMI_KCS_BMC_SERIO=m
@@ -2085,6 +2098,7 @@ CONFIG_HW_RANDOM=y
 CONFIG_HW_RANDOM_TIMERIOMEM=m
 CONFIG_HW_RANDOM_VIRTIO=m
 CONFIG_HW_RANDOM_HISI=m
+CONFIG_HW_RANDOM_PHYTIUM=m
 CONFIG_TCG_TIS=m
 CONFIG_TCG_TIS_SPI=m
 CONFIG_TCG_TIS_SPI_CR50=y
@@ -2169,6 +2183,8 @@ CONFIG_I2C_OCORES=m
 CONFIG_I2C_OMAP=m
 CONFIG_I2C_OWL=m
 CONFIG_I2C_PCA_PLATFORM=m
+CONFIG_I2C_PHYTIUM_PCI=m
+CONFIG_I2C_PHYTIUM_PLATFORM=m
 CONFIG_I2C_PNX=m
 CONFIG_I2C_PXA=m
 CONFIG_I2C_PXA_SLAVE=y
@@ -2336,7 +2352,10 @@ CONFIG_GPIO_PCA953X=m
 CONFIG_GPIO_PCF857X=m
 CONFIG_GPIO_DLN2=m
 CONFIG_GPIO_LP3943=m
+CONFIG_GPIO_PHYTIUM_PCI=m
 CONFIG_GPIO_VIPERBOARD=m
+CONFIG_W1=m
+CONFIG_W1_MASTER_PHYTIUM=y
 # CONFIG_POWER_RESET_BRCMSTB is not set
 CONFIG_POWER_RESET_GPIO=y
 CONFIG_POWER_RESET_GPIO_RESTART=y
@@ -2378,6 +2397,7 @@ CONFIG_SENSORS_ADT7475=m
 CONFIG_SENSORS_ASC7621=m
 CONFIG_SENSORS_ARM_SCMI=m
 CONFIG_SENSORS_ARM_SCPI=m
+CONFIG_SENSORS_PHYTIUM=m
 CONFIG_SENSORS_ATXP1=m
 CONFIG_SENSORS_DS620=m
 CONFIG_SENSORS_DS1621=m
@@ -2744,6 +2764,7 @@ CONFIG_VIDEO_MUX=m
 CONFIG_VIDEO_CADENCE_CSI2RX=m
 CONFIG_VIDEO_CADENCE_CSI2TX=m
 CONFIG_VIDEO_CAFE_CCIC=m
+CONFIG_VIDEO_PHYTIUM_JPEG=m
 CONFIG_VIDEO_XILINX=m
 CONFIG_VIDEO_XILINX_TPG=m
 CONFIG_SMS_SDIO_DRV=m
@@ -3347,6 +3368,8 @@ CONFIG_USB_DWC2_PCI=m
 CONFIG_USB_CHIPIDEA=m
 CONFIG_USB_CHIPIDEA_HOST=y
 CONFIG_USB_ISP1760=m
+CONFIG_USB_PHYTIUM=m
+CONFIG_USB_PHYTIUM_PCI=m
 CONFIG_USB_SERIAL=y
 CONFIG_USB_SERIAL_GENERIC=y
 CONFIG_USB_SERIAL_SIMPLE=m
@@ -3424,6 +3447,7 @@ CONFIG_USB_CXACRU=m
 CONFIG_USB_UEAGLEATM=m
 CONFIG_USB_XUSBATM=m
 CONFIG_USB_ISP1301=m
+CONFIG_USB_GADGET=m
 CONFIG_TYPEC=m
 CONFIG_TYPEC_TCPM=m
 CONFIG_TYPEC_FUSB302=m
@@ -3457,6 +3481,7 @@ CONFIG_MMC_REALTEK_USB=m
 CONFIG_MMC_TOSHIBA_PCI=m
 CONFIG_MMC_MTK=m
 CONFIG_MMC_SDHCI_XENON=m
+CONFIG_MMC_PHYTIUM_SDCI=m
 CONFIG_SCSI_UFSHCD=m
 CONFIG_SCSI_UFSHCD_PCI=m
 CONFIG_SCSI_UFSHCD_PLATFORM=m
@@ -3515,6 +3540,7 @@ CONFIG_INFINIBAND_ISER=m
 CONFIG_INFINIBAND_ISERT=m
 CONFIG_EDAC=y
 CONFIG_EDAC_XGENE=m
+CONFIG_EDAC_PHYTIUM=m
 CONFIG_RTC_CLASS=y
 CONFIG_RTC_DRV_ABB5ZES3=m
 CONFIG_RTC_DRV_ABX80X=m
@@ -3548,6 +3574,7 @@ CONFIG_DMADEVICES=y
 CONFIG_INTEL_IDMA64=m
 CONFIG_PL330_DMA=m
 CONFIG_PLX_DMA=m
+CONFIG_PHYTIUM_DDMA=y
 CONFIG_ASYNC_TX_DMA=y
 CONFIG_SW_SYNC=y
 CONFIG_UIO_CIF=m
@@ -3611,6 +3638,7 @@ CONFIG_COMMON_CLK_CDCE925=m
 # CONFIG_COMMON_CLK_XGENE is not set
 CONFIG_COMMON_CLK_PWM=m
 CONFIG_ARM_MHU=y
+CONFIG_PHYTIUM_MBOX=m
 CONFIG_PLATFORM_MHU=y
 CONFIG_PL320_MBOX=y
 CONFIG_HI6220_MBOX=m
@@ -3633,6 +3661,8 @@ CONFIG_EXTCON_PALMAS=m
 CONFIG_EXTCON_RT8973A=m
 CONFIG_EXTCON_SM5502=m
 CONFIG_ARM_PL172_MPMC=m
+CONFIG_IIO=m
+CONFIG_PHYTIUM_ADC=m
 CONFIG_NTB=y
 CONFIG_NTB_TRANSPORT=m
 CONFIG_PWM=y
@@ -3643,6 +3673,7 @@ CONFIG_PWM_LP3943=m
 CONFIG_PWM_PCA9685=m
 CONFIG_PWM_TWL=m
 CONFIG_PWM_TWL_LED=m
+CONFIG_PHYTIUM_IXIC=y
 CONFIG_IPACK_BUS=m
 CONFIG_BOARD_TPCI200=m
 CONFIG_SERIAL_IPOCTAL=m
@@ -3658,6 +3689,7 @@ CONFIG_PHY_QCOM_USB_HSIC=m
 CONFIG_PHY_SAMSUNG_USB2=m
 CONFIG_PHY_TUSB1210=m
 CONFIG_POWERCAP=y
+CONFIG_PHYTIUM_PMU=y
 CONFIG_STM=y
 CONFIG_STM_SOURCE_CONSOLE=y
 CONFIG_FPGA=m
-- 
2.20.1

